draft-ietf-quic-recovery.md

```diff
@@ -415,100 +414,88 @@ from when the first Initial was sent to when a Retry or a Version Negotiation
+flight but an acknowledgement does not seem forthcoming.  A PTO enables a
+connection to recover from loss of tail packets or acks, where acks of
+subsequent packets are not available to trigger ack-based loss detection.  The
+PTO algorithm used in QUIC implements the reliability functions of Tail Loss
+Probe {{?TLP=I-D.dukkipati-tcpm-tcp-loss-probe}}, RTO {{?RFC5681}} and F-RTO
+algorithms for TCP {{?RFC5682}}, and the computation is based on TCP's
+retransmission timeout period {{?RFC6298}}.
 
-The timer duration, or Probe Timeout (PTO), is set based on the following
-conditions:
+#### Computing PTO
 
-* PTO SHOULD be scheduled for max(1.5*SRTT+MaxAckDelay, kMinTLPTimeout)
+When the final ack-eliciting packet before quiescence is transmitted, the sender
+schedules a timer for the PTO period as follows:
 
-* If RTO ({{rto}}) is earlier, schedule a TLP in its place. That is,
-  PTO SHOULD be scheduled for min(RTO, PTO).
+PTO = max(smoothed_rtt + 4*rttvar + max_ack_delay, kGranularity)
 
-QUIC includes MaxAckDelay in all probe timeouts, because it assumes the ack
-delay may come into play, regardless of the number of packets outstanding.
-TCP's TLP assumes if at least 2 packets are outstanding, acks will not be
-delayed.
+kGranularity, smoothed_rtt, rttvar, and max_ack_delay are defined in
+{{ld-vars-of-interest}}.
 
-A PTO value of at least 1.5*SRTT ensures that the ACK is overdue.  The 1.5 is
-based on {{?TLP}}, but implementations MAY experiment with other constants.
+The PTO period is the amount of time that a sender ought to wait for an
+acknowledgement for a sent packet to be received.  This time period includes the
+estimated network roundtrip-time (smoothed_rtt), the variance in the estimate
+(4*rttvar), and max_ack_delay, to account for the maximum time by which a
+receiver might delay sending an acknowledgement.
 
-To reduce latency, it is RECOMMENDED that the sender set and allow the TLP timer
-to fire twice before setting an RTO timer. In other words, when the TLP timer
-expires the first time, a TLP packet is sent, and it is RECOMMENDED that the TLP
-timer be scheduled for a second time. When the TLP timer expires the second
-time, a second TLP packet is sent, and an RTO timer SHOULD be scheduled {{rto}}.
+There is no requirement on the clock granularity. If the PTO computation results
+in a value of zero, a sender MUST set the PTO value to kGranularity, to avoid
+the timer expiring immediately.
 
-A TLP packet SHOULD carry new data when possible. If new data is unavailable or
-new data cannot be sent due to flow control, a TLP packet MAY retransmit
-unacknowledged data to potentially reduce recovery time. Since a TLP timer is
-used to send a probe into the network prior to establishing any packet loss,
-prior unacknowledged packets SHOULD NOT be marked as lost when a TLP timer
-expires.
+A PTO timer is set on an ack-eliciting tail packet.  A sender may not know that
```

Comment:
Drop this first sentence and move the second sentence up with "When the final ack-eliciting packet"